src/publish/validatePackageDependencies.ts

```diff
@@ -4,30 +4,39 @@ import { shouldPublishPackage } from './shouldPublishPackage';
 /**
  * Validate no private package is listed as package dependency for packages which will be published.
  */
-export function validatePackageDependencies(bumpInfo: BumpInfo): boolean {
+export function validatePackageDependencies(bumpInfo: BumpInfo, quiet?: boolean): boolean {
   let hasErrors: boolean = false;
   const { modifiedPackages, newPackages, packageInfos } = bumpInfo;
 
   const packagesToValidate = [...modifiedPackages, ...newPackages];
-  let allDeps: string[] = [];
-  packagesToValidate.forEach(pkg => {
+  /** Mapping from dep to all validated packages that depend on it */
+  const allDeps: { [dep: string]: string[] } = {};
+  for (const pkg of packagesToValidate) {
     const { publish, reasonToSkip } = shouldPublishPackage(bumpInfo, pkg);
     if (!publish) {
-      console.log(`Skipping package dep validation - ${reasonToSkip}`);
-      return;
+      if (!quiet) {
+        console.log(`Skipping package dep validation - ${reasonToSkip}`);
```

Comment:
So, I know this is an annoyance... but can we isolate the change in this PR without changing this function's signature?

I have in mind an overhaul of the logging of beachball to use something a lot like what's used inside `lage` where I can do a more flexible "reporter" infra (with a default that can have error levels).

Reply:
#484 Is the issue I created for this kind of work.